addons/base_import/models/base_import.py

```diff
@@ -444,6 +444,9 @@ def _try_match_date_time(self, preview_values, options):
         # Or a date/datetime if it matches the pattern
         date_patterns = [options['date_format']] if options.get(
             'date_format') else []
+        user_date_format = self.env['res.lang'].search([('code', '=', self.env.user.lang)]).date_format
```

Comment:
Should probably use _lang_get here (/cc @mart-e).

Reply:
In stable keep in mind it will fallback if the lang is not active but it should not happen on `self.env.user` so yes, use `_lang_get`